tweet_recommendations/embeddings/sif_embedding.py

```diff
@@ -57,51 +71,74 @@ def fit(self, x: pd.DataFrame, y: Optional[pd.DataFrame] = None,
+                'contain column "lemmas".'
+
+        emb = self._get_weighted_average_embeddings(x)
 
         self._pc = self._compute_pc(emb, fit_params.get("random_state", None))
 
         return self
 
-    def transform(self, x: Union[Tuple[Tuple[str, ...]], Tuple[str, ...]], **kwargs) -> np.ndarray:
+    def transform(self, x: Union[Tuple[Tuple[str, ...]], Tuple[str, ...], pd.DataFrame], **kwargs) -> np.ndarray:
         """
         For a given tweet represented as a list of lemmas recommends hashtags.
         :param x: list of list of str or list of str. If first argument of x is a list is str, it is assumed that list
             contains already lemmatized text. If single str is present as first element, it is assumed
-            that lemmatization has to be performed.
+            that lemmatization has to be performed. A DataFrame can be passed if it contains lemmatized sentences with
+            its corresponding embeddings.
         :return: np.ndarray of sentences embeddings.
         """
-        lemmatized = []
-        if isinstance(x, pd.DataFrame):
-            lemmatized = x["lemmas"].to_list()
-        elif isinstance(x[0], str):
-            for i, xi in enumerate(x):
-                lemmatized.append(get_wcrft2_results_for_text(xi))
-        elif isinstance(x[0], list):
-            for i, xi in enumerate(x):
-                lemmatized.append(xi)
-        output = self._get_sif_embedding(lemmatized)
+        if self.keyed_vector_model:
+            lemmatized = []
+            if isinstance(x, pd.DataFrame):
+                lemmatized = x["lemmas"].to_list()
+            elif isinstance(x[0], str):
+                for i, xi in enumerate(x):
+                    lemmatized.append(get_wcrft2_results_for_text(xi))
+            elif isinstance(x[0], list):
+                for i, xi in enumerate(x):
+                    lemmatized.append(xi)
+            data = lemmatized
+        else:
+            data = x
+        output = self._get_sif_embedding(data)
         return np.asarray(output)
 
-    def _get_weighted_average_embeddings(self, sentences: List[List[str]]) -> np.ndarray:
+    def _get_weighted_average_embeddings(self, data: pd.DataFrame) -> np.ndarray:
         """
         Compute the weighted average vectors for given tweet content lemmas.
-        :param sentences: List of tweet content lemmas list.
+        :param data: DataFrame which should contain tweet text lemmas.
         :return: Weighted average of tweet embeddings.
         """
+        sentences = data["lemmas"].values.tolist()
         n_samples = len(sentences)
-        emb = np.zeros((n_samples, self.keyed_vector_model.vector_size))
+        if self.keyed_vector_model:
+            embedding_size = self.keyed_vector_model.vector_size
+        else:
+            embedding_size = len(data[self.embedding_name].iloc[0])
+        emb = np.zeros((n_samples, embedding_size))
```

Comment:
czy to jest w ogóle wykorzystywane? nie widzę tego...

Reply:
To jest w sumie tylko do zdefiniowania rozmiaru macierzy wynikowej. Dodatkowo jeśli się trafi, że mamy pustą treść tweeta (a może się tak zdażyć) to na odpowiednim miejscu z automatu mamy embedding o zerowych wartościach. (po to jest ten if w 129 linijce)